swap_meet/vendor.py

```diff
@@ -46,11 +43,23 @@ def swap_first_item(self, friend):
         # remove item from friend.inventory and add it to self.inventory
 
     def get_best_by_category(self, category):
-        # loop through the self.inventory
         best_item = None
         for item in self.get_by_category(category):
-            if best_item == None:
+            if best_item is None:
                 best_item = item
             elif item.condition > best_item.condition:
                 best_item = item
-        return best_item
\ No newline at end of file
+        return best_item
+
+    # this function will swap with other, get my priority, get their priority
+    # def swap_best_by_category(self, other, my_priority, their_priority):
+    # can use get_best_by_category and swap_first_item (since it's the best by category)
+
+    def swap_best_by_category(self, other, my_priority, their_priority):
+        my_best_item = self.get_best_by_category(their_priority)
+        their_best_item = other.get_best_by_category(my_priority)
+        if their_best_item is None or my_best_item is None:
+            return False
+        else:
+            self.swap_items(other, my_best_item, their_best_item)
+            return True
```

Comment:
Great job! Really nice re-use of your earlier methods here. We don't need the `else` here, can you consider why?

Reply:
Instead of using else I could just use another if statement. Since line 62 returns False, it exits the function. I realized that just now :)